Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp

```diff
@@ -1899,26 +1898,44 @@ void NetworkResourceLoader::setWorkerStart(MonotonicTime value)
 void NetworkResourceLoader::startWithServiceWorker()
 {
     LOADER_RELEASE_LOG("startWithServiceWorker:");
+
+    auto newRequest = ResourceRequest { originalRequest() };
+#if ENABLE(CONTENT_FILTERING_IN_NETWORKING_PROCESS)
+    if (!startContentFiltering(newRequest))
+        return;
+#endif
+
     ASSERT(!m_serviceWorkerFetchTask);
-    m_serviceWorkerFetchTask = m_connection->createFetchTask(*this, originalRequest());
+    m_serviceWorkerFetchTask = m_connection->createFetchTask(*this, newRequest);
     if (m_serviceWorkerFetchTask) {
         LOADER_RELEASE_LOG("startWithServiceWorker: Created a ServiceWorkerFetchTask (fetchIdentifier=%" PRIu64 ")", m_serviceWorkerFetchTask->fetchIdentifier().toUInt64());
         return;
     }
 
-    serviceWorkerDidNotHandle(nullptr);
+    if (requestIsForServiceWorkerOnly())
+        return;
+
+    startRequest(newRequest);
+}
+
+bool NetworkResourceLoader::requestIsForServiceWorkerOnly()
```

Comment:
It may be named as `checkServiceWorkersMode` or `abortIfServiceWorkersOnly`, since `requestIsForServiceWorkerOnly` sounds like it only returns the value of `m_parameters.serviceWorkersMode == ServiceWorkersMode::Only`.

Reply:
> It may be named as `checkServiceWorkersMode` or `abortIfServiceWorkersOnly`, since `requestIsForServiceWorkerOnly` sounds like it only returns the value of `m_parameters.serviceWorkersMode == ServiceWorkersMode::Only`.

Fixed!